orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/Stage.java

```diff
@@ -67,6 +67,10 @@ public Stage(T execution, String type) {
    */
   private String name;
 
+  public String getName() {
+    return name != null ? name : type;
```

Comment:
This might be redundant ... I had a contrived scenario and was getting blasted by a null `Stage.getName()` which had a `@NonNull`.